src/pyodide/python-entrypoint-helper.ts

```diff
@@ -232,17 +232,30 @@ async function initPyInstance(
   return res as PyModule;
 }
 
+// https://developers.cloudflare.com/workers/runtime-apis/rpc/reserved-methods/
+const SPECIAL_HANDLER_NAMES = ['fetch', 'connect'];
+const SPECIAL_DO_HANDLER_NAMES = [
+  'alarm',
+  'webSocketMessage',
+  'webSocketClose',
+  'webSocketError',
+];
+
 function makeEntrypointProxyHandler(
-  pyInstancePromise: Promise<PyModule>
+  pyInstancePromise: Promise<PyModule>,
+  isDurableObject: boolean
 ): ProxyHandler<any> {
   return {
     get(target, prop, receiver): any {
       if (typeof prop !== 'string') {
         return Reflect.get(target, prop, receiver);
       }
+
       // Proxy calls to `fetch` to methods named `on_fetch` (and the same for other handlers.)
-      const isKnownHandler = SUPPORTED_HANDLER_NAMES.includes(prop);
-      if (isKnownHandler) {
+      const isKnownHandler = SPECIAL_HANDLER_NAMES.includes(prop);
+      const isKnownDoHandler =
+        SPECIAL_DO_HANDLER_NAMES.includes(prop) && isDurableObject;
+      if (isKnownHandler || isKnownDoHandler) {
```

Comment:
It's a bit more efficient to combine it into one condition (so we get short circuit evaluation):
```suggestion
      if (SPECIAL_HANDLER_NAMES.includes(prop) || isDurableObject && SPECIAL_DO_HANDLER_NAMES.includes(prop)) {
```
If not this, at least it should be `isDurableObject && SPECIAL_DO_HANDLER_NAMES.includes(prop)`.

Reply:
Yeah... I prefer the clearer code in this case. But I will flip the order of `isKnownDoHandler`.